tests/testInitialSetupConsent.php

```diff
@@ -204,14 +204,70 @@ public function testThatNewConnectionIsCreatedAndFailedClientGrantRedirects() {
 		$this->assertCount( 1, self::$error_log->get() );
 	}
 
+	/**
+	 * Test that an connection is created and the Client Grant fails.
+	 */
+	public function testThatNewConnectionIsCreatedWithExistingMigrationToken() {
+		$this->startHttpMocking();
+		$this->startRedirectHalting();
+
+		$setup_consent = new WP_Auth0_InitialSetup_Consent( self::$opts );
+		$test_token    = implode( '.', [ uniqid(), uniqid(), uniqid() ] );
+
+		self::$opts->set( 'client_signing_algorithm', 'HS256' );
+		self::$opts->set( 'migration_token', 'TEST_MIGRATION_TOKEN' );
+
+		// Mock consecutive HTTP calls.
+		$this->http_request_type = [
+			// Successful client creation.
+			'success_create_client',
+			// Get en existing connection enabled for this client.
+			'success_get_connections',
+			// Connection updated successfully.
+			'success_update_connection',
+			// Connection created successfully.
+			'success_create_connection',
+			// Client grant created successfully.
+			'success_create_client_grant',
+		];
+
+		$caught_redirect = [];
+		try {
+			$setup_consent->callback_with_token( 'test-wp.auth0.com', $test_token, 'social' );
+		} catch ( Exception $e ) {
+			$caught_redirect = unserialize( $e->getMessage() );
+		}
+
+		$this->assertNotEmpty( $caught_redirect );
+		$this->assertEquals( 302, $caught_redirect['status'] );
+
+		$redirect_url = parse_url( $caught_redirect['location'] );
+
+		$this->assertEquals( '/wp-admin/admin.php', $redirect_url['path'] );
+		$this->assertContains( 'page=wpa0-setup', $redirect_url['query'] );
+		$this->assertContains( 'step=2', $redirect_url['query'] );
+		$this->assertContains( 'profile=social', $redirect_url['query'] );
+
+		$this->assertEquals( 'TEST_CLIENT_ID', self::$opts->get( 'client_id' ) );
+		$this->assertEquals( 'TEST_CLIENT_SECRET', self::$opts->get( 'client_secret' ) );
+		$this->assertEquals( 1, self::$opts->get( 'db_connection_enabled' ) );
+		$this->assertEquals( 'TEST_CREATED_CONN_ID', self::$opts->get( 'db_connection_id' ) );
+		$this->assertEquals( 'TEST_MIGRATION_TOKEN', self::$opts->get( 'migration_token' ) );
```

Comment:
Please add a test that asserts that when this token is not set (Empty) a random token (not empty, length >x, etc) is being used. I couldn't find it on this file.

Reply:
See the added line in the test above (which is uses that process)